core/src/saros/net/xmpp/contact/IContactsUpdate.java

```diff
@@ -0,0 +1,28 @@
+package saros.net.xmpp.contact;
+
+import java.util.Optional;
+import saros.net.xmpp.XMPPConnectionService;
+
+/** Interface for receiving Contact Updates from {@link XMPPConnectionService}. */
+public interface IContactsUpdate {
+
+  /** List of possible update types. */
+  enum UpdateType {
+    ADDED,
+    CONNECTED,
+    NOT_CONNECTED,
+    NICKNAME_CHANGED,
+    REMOVED,
+    STATUS,
+    SAROS_SUPPORT,
```

Comment:
@srossbach How would you like to encapsulate the smack stuff and separating it from Saros?
Separating saros specific feature detection from other logic or what is you proposal?

Reply:
If you look at https://github.com/saros-project/saros/pull/630/files#diff-a74ef4faaee4f6df31ff9d803a0de626R133 it is quite clear how the support is determined. What makes me wonder is why the XMPPContact class does not provide any method to query the supported features for all / a given presence.

the static utility method would look like this

```
private static JID getSarosPresence(XMPPContact contact) {
   for (Presence p : contact.getPresences())
          if (p.getFeatures().contains(THE_MAGIC_ID))
               return p.getFqJid()

   return null
}
```

Reply:
From my point of view, we basically don't care about resources outside the xmpp package.
XMPPContact sorts its known resources to select just the important one for us, the one with Saros Support. So removing this, removes the sorting and again other logic has to decide which resource is important.
If a Contact has 7 Pidign Clients online and one Saros Client we look at the Saros Client.
When Saros would supports setting its own Contact to away, I would say its enough to present this Contact as away even with some other Clients online.

Reply:
You just moved UI display logic into the network layer and now telling me this is business logic ? 

If the UI was coded in such a way that it would display all presences for a resource would the moved code provides the same logic ? I do not get the reason why you hide data that cannot be accessed and in turn decides which data is handed out, e.g as I understand the code you do not have access to the results of the service discovery.

Reply:
I'm telling that we can have an abstraction of all the xmpp resource stuff because we don't need to know. 
>  I do not get the reason why you hide data [...]

Abstraction makes the rest of the code base easier and what is your use-case for knowing resources?

> as I understand the code you do not have access to the results of the service discovery.

What do you mean?
https://github.com/saros-project/saros/blob/38293504234bb648a574663dae167e7d9453b569/core/src/saros/net/xmpp/contact/DiscoveryService.java#L67

Reply:
>From my point of view, we basically don't care about resources outside the xmpp package.

And because of this a (possible bare) JID has to be transformed to a  FQ-JID in the Session negotiation although the FQ-JID is already known by the UI? Sorry this is not convenience, this is just lazy and bad design, it is like trying to connect to a broadcast IP address.

IMO the whole Core should reject bare JIDs.

Reply:
I think we should schedule a call for this topic as it seems to me that my point isn't clear.
Did you get my mail about the review this Friday?

But to answer this already, my point is to remove the usage of jids as an identifier for a Contact. So the next stages are that the core shouldn't accept jids at all. Contacts leave the core to the ui, and the ui calls the core with theses contacts. At the end jids don't leave the net package.